src/test/rgw/rgw_multi/tests.py

```diff
@@ -1766,6 +1711,301 @@ def bucket_keys_eq(zone1, zone2, bucket_name):
+        iam.put_group_policy(GroupName=name, PolicyName='Allow', PolicyDocument=inline_policy)
+        iam.attach_group_policy(GroupName=name, PolicyArn=managed_policy_arn)
+        iam.add_user_to_group(GroupName=name, UserName=name)
+        # create role, role policy, managed policy
+        iam.create_role(RoleName=name, AssumeRolePolicyDocument=json.dumps({'Version': '2012-10-17', 'Statement': [{'Effect': 'Allow', 'Principal': {'AWS': 'arn:aws:iam:::user/testuser'}, 'Action': ['sts:AssumeRole']}]}))
+        iam.put_role_policy(RoleName=name, PolicyName='Allow', PolicyDocument=inline_policy)
+        iam.attach_role_policy(RoleName=name, PolicyArn=managed_policy_arn)
+        # TODO: test oidc provider
+        #iam.create_open_id_connect_provider(ClientIDList=['clientid'], ThumbprintList=['3768084dfb3d2b68b7897bf5f565da8efEXAMPLE'], Url=f'http://{name}.example.com')
+
+    realm_meta_checkpoint(realm)
+
+    # check that all users/groups/roles are equal across all zones
+    for source_conn, target_conn in combinations(zonegroup_conns.zones, 2):
+        if target_conn.zone.has_roles():
+            check_roles_eq(source_conn, target_conn)
+            check_users_eq(source_conn, target_conn)
+            check_groups_eq(source_conn, target_conn)
+            check_oidc_providers_eq(source_conn, target_conn)
+
+    for source_conn in zonegroup_conns.rw_zones:
+        iam = source_conn.iam_conn
+        name = source_conn.name
+
+        #iam.delete_open_id_connect_provider(OpenIDConnectProviderArn=f'arn:aws:iam::RGW11111111111111111:oidc-provider/{name}.example.com')
+
+        iam.detach_role_policy(RoleName=name, PolicyArn=managed_policy_arn)
+        iam.delete_role_policy(RoleName=name, PolicyName='Allow')
+        iam.delete_role(RoleName=name)
+
+        iam.remove_user_from_group(GroupName=name, UserName=name)
+        iam.detach_group_policy(GroupName=name, PolicyArn=managed_policy_arn)
+        iam.delete_group_policy(GroupName=name, PolicyName='Allow')
+        iam.delete_group(GroupName=name)
+
+        iam.detach_user_policy(UserName=name, PolicyArn=managed_policy_arn)
+        iam.delete_user_policy(UserName=name, PolicyName='Allow')
+        key_id = iam.list_access_keys(UserName=name)['AccessKeyMetadata'][0]['AccessKeyId']
+        iam.delete_access_key(UserName=name, AccessKeyId=key_id)
+        iam.delete_user(UserName=name)
+
+    realm_meta_checkpoint(realm)
+
+    # check that all users/groups/roles are equal across all zones
+    for source_conn, target_conn in combinations(zonegroup_conns.zones, 2):
+        if target_conn.zone.has_roles():
+            check_roles_eq(source_conn, target_conn)
+            check_users_eq(source_conn, target_conn)
+            check_groups_eq(source_conn, target_conn)
+            check_oidc_providers_eq(source_conn, target_conn)
+
+
+# TODO: move sync policy and endpoint tests into separate test file
+# fix data sync init tests
+'''
+def test_bucket_reshard_index_log_trim():
```

Comment:
instead of commenting these out, could we use a nose attr like "fails_on_rgw" and filter that out when running teuthology?

Reply:
that way we don't need to move test cases around when we want to disable/reenable them

Reply:
sure @cbodley , that sounds better. 
here are two separate runs:
https://qa-proxy.ceph.com/teuthology/smanjara-2024-04-15_09:53:10-rgw:multisite-wip-shilpa-multisite-tests-distro-default-smithi/7657071/teuthology.log
https://qa-proxy.ceph.com/teuthology/smanjara-2024-04-15_09:53:28-rgw:multisite-wip-shilpa-multisite-tests-distro-default-smithi/7657074/teuthology.log

in both of them, test_topic_notification_sync() fails. could you take a look please? everything else passes with the two-zone configuration.

Reply:
currently, I have to pass in the configuration to use two-zone using ` --filter` in teuthology. how do we turn the other two configurations, two-zonegoup and three-zones off so that two-zone is run by default?

Reply:
> how do we turn the other two configurations, two-zonegoup and three-zones off so that two-zone is run by default?

you can remove or rename their yaml files under qa/suites/rgw/multisite/realms/

for example, when i disabled valgrind in the multisite suite, i renamed qa/suites/rgw/multisite/valgrind.yaml to valgrind.yaml.disabled to prevent teuthology from selecting it for jobs

Reply:
ok yeah will rename thanks!

Reply:
another question. how do you filter out based on nose attrs in teuthology? could you share a command if you have it handy?